uber/site_sections/registration.py

```diff
@@ -672,20 +672,23 @@ def redeem_merch_discount(self, session, badge_num, apply=''):
     @unrestricted
     @check_atd
     @check_if_can_reg
-    def register(self, session, message='', **params):
+    def register(self, session, message='', error_message='', **params):
         params['id'] = 'None'
         attendee = session.attendee(params, restricted=True, ignore_csrf=True)
-        if 'first_name' in params:
+        error_message = check_pii_consent(params, attendee) or error_message
+        if not error_message and 'first_name' in params:
             if not attendee.payment_method and (not c.BADGE_PRICE_WAIVED or c.BEFORE_BADGE_PRICE_WAIVED):
-                message = 'Please select a payment type'
+                error_message = 'Please select a payment type'
             elif attendee.payment_method == c.MANUAL and not re.match(c.EMAIL_RE, attendee.email):
-                message = 'Email address is required to pay with a credit card at our registration desk'
+                error_message = 'Email address is required to pay with a credit card at our registration desk'
             elif attendee.badge_type not in [badge for badge, desc in c.AT_THE_DOOR_BADGE_OPTS]:
-                message = 'No hacking allowed!'
+                error_message = 'No hacking allowed!'
             else:
-                message = check(attendee)
+                error_message = check(attendee)
 
-            if not message:
+            if not error_message and c.BADGE_PROMO_CODES_ENABLED and 'promo_code' in params:
+                error_message = session.add_promo_code_to_attendee(attendee, params.get('promo_code'))
```

Comment:
Surprisingly, at the door registration wasn't applying promo codes. Fixed that bug!